ProjectSolenodon/Assets/Scripts/LevelManager.cs

```diff
@@ -0,0 +1,61 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using UnityEngine;
+
+/// <summary>
+/// The LevelManager class is used to build scenes that associated to different levels
+/// by instantiating needed game objects.
+/// It also has the ability to serialize and deserialize LevelData objects.
+/// </summary>
+public class LevelManager : MonoBehaviour
+{
+    public LevelData levelData;
+    public GameObject registerCard;
+    public GameObject stackCard;
+    public GameObject queueCard;
+    public GameObject heapCard;
+
+    private Transform LevelHolder;
+
+    void DeserializeLevelData(int levelId)
+    {
+        string levelPath = "Assets/Resources/LevelSaves/level-" + levelId + ".json";

```

Comment:
I don't know if this is possible in Unity, but in Unreal it is often possible to set class data from the editor. 

The motivation for that would be to avoid the possibility of your code failing silently when files are moved, renamed, or deleted. If set through the editor, you alerted to the error.

Reply:
You can set class data via the editor. Once the script is linked, the parent asset will have an additional drop down in the script and you can just drag and drop which asset you want to be linked with that part of the script.

Reply:
Leaving this for now, since the JSON file we want to read can change depending which puzzle has been chosen. If a better way to do this comes up in the future we can readdress this.